### ISSUE TITLE
noisefigure/noisefigure.cpp:183:26: error: ‘barycentric_rational’ is not a member of ‘boost::math’

### ISSUE THREAD

**Issue by blshkv** [f4exb/sdrangel#1087]
I found an old bug report (https://github.com/f4exb/sdrangel/issues/943) but the problem seems back with ```boost-1.77.0```

https://github.com/pentoo/pentoo-overlay/issues/1054

the error: https://gist.github.com/asarubbo/28c0b567f564f17db2026f1e9b74eccd
the build log and system details: https://gist.github.com/asarubbo/38dc24e1b44ac4c96b5a5aa42eccc574

A workaround was proposed here:
https://stackoverflow.com/questions/13955627/boost-math-special-functions-compilation-error-when-using-boost-for-tr1

```
We were able to work around this by replacing

#include<boost/random.hpp>
with the more specific to our needs:

#include<boost/random/mersenne_twister.hpp>
#include<boost/random/uniform_int.hpp>
#include<ctime>
thus leaving out the problematic headers.
```

**Comment by f4exb** [comment]
On most systems the package installation uses earlier versions of Boost that do not exhibit the issue. Ex: 1.74 with Ubuntu 21.10. Now Boost 1.78 is out. I would be curious to know if this has been fixed or not in this version.

**Comment by f4exb** [comment]
This seems related to very old versions of gcc.

**Comment by blshkv** [comment]
So again, you can find all related details in the log:
sys-devel/gcc:            11.2.0::gentoo
sys-devel/gcc-config:     2.5-r1::gentoo

**Comment by srcejon** [comment]
#943 is a different issue. The constructor used wasn't available in the earlier version of boost that the user had.

I've just tried boost 1.77 and 1.78 on Windows, and I do see a compilation failure. I don't think the proposed workaround is relevant though, as we don't actually include boost/random.hpp.

It appears that in boost 1.77 boost::math::barycentric_rational has moved to boost::math::interpolators::barycentric_rational

https://www.boost.org/doc/libs/1_76_0/libs/math/doc/html/math_toolkit/barycentric.html
https://www.boost.org/doc/libs/1_77_0/libs/math/doc/html/math_toolkit/barycentric.html

**Comment by f4exb** [comment]
Implemented in v6.17.7
